crates/bevy_audio/src/audio_source.rs

```diff
@@ -18,22 +18,92 @@ impl AsRef<[u8]> for AudioSource {
 }
 
 /// Loads mp3 files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "mp3")]
 #[derive(Default)]
 pub struct Mp3Loader;
 
+#[cfg(feature = "mp3")]
 impl AssetLoader for Mp3Loader {
-    fn load(&self, bytes: &[u8], load_context: &mut LoadContext) -> BoxedFuture<Result<()>> {
-        load_context.set_default_asset(LoadedAsset::new(AudioSource {
-            bytes: bytes.into(),
-        }));
-        Box::pin(async move { Ok(()) })
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
     }
 
     fn extensions(&self) -> &[&str] {
-        &["mp3", "flac", "wav", "ogg"]
+        &["mp3"]
     }
 }
 
+/// Loads flac files as [AudioSource] [Assets](bevy_asset::Assets)
+#[cfg(feature = "flac")]
+#[derive(Default)]
+pub struct FlacLoader;
+
+#[cfg(feature = "flac")]
+impl AssetLoader for FlacLoader {
+    fn load<'a>(
+        &'a self,
+        bytes: &'a [u8],
+        load_context: &'a mut LoadContext,
+    ) -> BoxedFuture<'a, Result<()>> {
+        load(bytes, load_context)
+    }
+
+    fn extensions(&self) -> &[&str] {
+        &["flac"]
```

Comment:
Instead of adding multiple asset loaders, could you do something like:
```rust
        &[
            #[cfg(feature = "mp3")]
            "mp3",
            #[cfg(feature = "flac")]
            "flac",
            #[cfg(feature = "wav")]
            "wav",
            #[cfg(feature = "vorbis")]
            "ogg",
        ]
```

Reply:
And then also have a similar approach with the `Texture` assets?

Reply:
I didn't know this was possible and this is a much better solution.